app.model.js

```diff
@@ -26,8 +26,12 @@ exports.fetchAllArticles = ()=>{
 }
 
 exports.fetchAllArticleComments = (article_id)=>{
-    
-    return db.query(`SELECT * FROM comments WHERE article_id=$1 ORDER BY created_at DESC`, [article_id])
+
+    return db.query(`SELECT * FROM articles WHERE article_id=$1`, [article_id]).then(({rows})=>{
+        if(!rows[0]) return Promise.reject(404)
+        return Promise.resolve(article_id)
```

Comment:
A nice touch but for what it's worth, it's not necessary to resolve the Promise

Reply:
There's a couple of ways you could improve this. For one thing, you already have a model function that calls an article by id, so that could be reused here. Also, if we need to handle two promises, a Promise.all could come in handy. If we made this change, the reject you have on line 11 would need to go inside the fetch article by id function. From what I can see, what you have works, but it is a little bit repetitive of previous code

Reply:
noted